crates/sqlbuiltins/src/functions/table/object_store.rs

```diff
@@ -42,10 +44,40 @@ pub const READ_JSON: ObjScanTableFunc = ObjScanTableFunc(FileType::JSON, "read_n
 #[derive(Debug, Clone)]
 pub struct ObjScanTableFunc(FileType, &'static str);
 
-#[async_trait]
-impl TableFunc for ObjScanTableFunc {
-    fn runtime_preference(&self) -> RuntimePreference {
-        RuntimePreference::Unspecified
+impl BuiltinFunction for ObjScanTableFunc {
+    fn name(&self) -> &'static str {
+        self.1
+    }
+    fn function_type(&self) -> FunctionType {
+        FunctionType::TableReturning
+    }
+    fn sql_example(&self) -> Option<String> {
+        fn build_example(extension: &str) -> String {
+            format!("{ext}_scan('./my_data.{ext}')", ext = extension)
+        }
+        Some(build_example(self.0.to_string().as_str()))
+    }
+    fn description(&self) -> Option<String> {
+        fn build_description(extension: &str) -> String {
+            format!(
+                r#"
```

Comment:
<question>: in future this gets replaced by the online help you're building from our docs content?